theseus_gui/src/pages/instance/Index.vue

```diff
@@ -41,23 +60,76 @@
   </div>
 </template>
 <script setup>
-import { BoxIcon, SettingsIcon, FileIcon, Button, Avatar, Card, Promotion } from 'omorphia'
+import { BoxIcon, SettingsIcon, FileIcon, XIcon, Button, Avatar, Card, Promotion } from 'omorphia'
 import { PlayIcon, OpenFolderIcon } from '@/assets/icons'
 import { get, run } from '@/helpers/profile'
+import {
+  get_all_running_profile_paths,
+  get_uuids_by_profile_path,
+  kill_by_uuid,
+} from '@/helpers/process'
+import { process_listener } from '@/helpers/events'
 import { useRoute } from 'vue-router'
-import { shallowRef } from 'vue'
+import { shallowRef, ref, onUnmounted } from 'vue'
 import { convertFileSrc } from '@tauri-apps/api/tauri'
+import { open } from '@tauri-apps/api/dialog'
 import { useBreadcrumbs } from '@/store/breadcrumbs'
 
-const breadcrumbs = useBreadcrumbs()
-
 const route = useRoute()
+const breadcrumbs = useBreadcrumbs()
 const instance = shallowRef(await get(route.params.id))
+
 breadcrumbs.setName('Instance', instance.value.metadata.name)
 breadcrumbs.setContext({
   name: instance.value.metadata.name,
   link: route.path,
 })
+
+const uuid = ref(null)
+const playing = ref(false)
+const loading = ref(false)
+
+const startInstance = async () => {
+  loading.value = true
+  uuid.value = await run(route.params.id)
+  loading.value = false
+  playing.value = true
+}
+
+const checkProcess = async () => {
+  const runningPaths = await get_all_running_profile_paths()
+  if (runningPaths.includes(instance.value.path)) {
+    playing.value = true
+    return
+  }
+
+  playing.value = false
+  uuid.value = null
+}
+
+await checkProcess()
+
+const stopInstance = async () => {
+  playing.value = false
+
+  try {
+    if (!uuid.value) {
+      const uuids = await get_uuids_by_profile_path(instance.value.path)
+      uuids.forEach(async (u) => await kill_by_uuid(u))
+    } else await kill_by_uuid(uuid.value)
+  } catch (err) {
+    // Theseus currently throws:
+    //  "Error launching Minecraft: Minecraft exited with non-zero code 1" error
+    // For now, we will catch and just warn
+    console.warn(err)
+  }
+}
+
+const unlisten = await process_listener((e) => {
+  if (e.event === 'Finished') playing.value = false
+})
```

Comment:
Won't this not work if you have multiple instances running? it should check the PID to make sure it's the right one